CMakeLists.txt

```diff
@@ -106,7 +106,18 @@ endif()
 #----------------------------------------------------------------------------
 # Find MPI and set some flags so that FC and CC can point to gfortran and gcc
 #----------------------------------------------------------------------------
-find_package(MPI REQUIRED)
+
+# If the user passes FC=mpif90 etc. check and prefer that location
+get_filename_component( FTN_MPI_DIR "${CMAKE_Fortran_COMPILER}"
+  REALPATH )
+get_filename_component( C_MPI_DIR "${CMAKE_C_COMPILER}"
+  REALPATH )
+set ( MPI_HOME "${MPI_HOME}" "${FTN_MPI_DIR}/.." "${C_MPI_DIR}/.." )
```

Comment:
Put compiler paths (up one directory, since they're probably in `bin`) in the MPI search list